students/mayc4t/lesson06/mailroom4_test.py

```diff
@@ -0,0 +1,214 @@
+#!/usr/bin/env python3
+
+import mailroom4
+import unittest
+from unittest.mock import patch
+import os
+
+
+class TestMailRoom(unittest.TestCase):
+    
+    # Create "known values" or equivalent to setUp(self)
+
+    def setUp(self):
+        """Test SetUp
+            Set Up the donor_db before each test
+            This is called before every single test to initialize donor_db
+        """
+        mailroom4.init_db()
+
+    # Clean up after everytime testing
+    def tearDown(self):
+        """
+            Tear Down
+            Clear up all the extra .txt thank you note get sent
+        """
+        for file in os.listdir( os.getcwd()):
+             if '.txt' in file:  os.remove(file)
+
+    def test_is_donor_in_list(self):
+        """ 
+            Test to see if "REAL" donor is in the list regardless how it was written
+        """
+        # test donor is already in list
+        expected = (True, "Kate Spade")
+        result = mailroom4.is_donor_in_list("Kate Spade")
+        self.assertTrue( expected, result)
+        
+        # test donor NOT already in list
+        expected = (False, "New Person" )
+        result = mailroom4.is_donor_in_list("New Person")
+        self.assertTrue( expected, result)
+
+
+    def test_enter_dn_name(self):
+        """
+        Test Enter Donor Name 
+        """
+        def input_mock(self):
+            return 'Michael Kors'
```

Comment:
You should be able to use the `mock.patch` `side_effect` parameter to mock input like this